alibi/explainers/similarity/grad.py

```diff
@@ -91,25 +148,25 @@ def _preprocess_args(
 
         Parameters
         ----------
-        X:
+        X
             Input data requiring formatting.
-        Y:
-            target data or function requiring formatting.
+        Y
+            Target data or function requiring formatting.
 
         Returns
         -------
-        X:
+        X
             Input data formatted for explain method.
-        Y:
+        Y
             Target data formatted for explain method.
 
         """
-        X = self._match_shape_to_data(X, 'x')
+        X = self._match_shape_to_data(X, 'X')
         if isinstance(X, np.ndarray):
             X = self.backend.to_tensor(X)
 
-        if self.task == 'regression' and Y is None:
-            err_msg = 'Regression task requires a target value. `Y` must be provided, either as a value or a function.'
+        if self.task == Task.REGRESSION and Y is None:
+            err_msg = "Regression task requires a target value. 'Y' must be provided, either as a value or a function."
```

Comment:
OK so we need to decide if we want the custom callable option or not. Is the use case you had in mind here for multi-output regression?

Reply:
I think in retrospect I added `Y` as a custom callable by mistake. I can't think how it would apply to the multi regression case. Currently, multi-regression just requires the correct shape label passed as the `Y`.  I think I'll remove!

See [here](https://github.com/SeldonIO/alibi/pull/577#discussion_r841673079) for related.